tests/test_workers/test_tasks/test_opm_operations.py

```diff
@@ -897,9 +923,17 @@ def test_opm_registry_add_fbc_fragment(
             ]
         )
         assert mock_cpt.call_count == 2
+        # Assert deprecations were removed correctly
+        assert not os.path.exists(deprecation_file)
+        assert not operator_deprecation_dir.check()
+        assert deprecations_dir.check(dir=True)
     else:
         assert mock_cpt.call_count == 1
         assert mock_orr.call_count == 0
+        # Assert deprecations are still present
+        assert os.path.exists(deprecation_file)
```

Comment:
For the test, this is fine.. but in the practical situation if the package is not present in the db, we assume that it's not the first time run of an opt in package, and its deprecation file should have been already removed in a previous request.

Wondering if we should remove the deprecation file for each fbc request of the package if the file is present, irrespective of it was present in db or not, wdyt?

Reply:
> Wondering if we should remove the deprecation file for each fbc request of the package if the file is present, irrespective of it was present in db or not, wdyt?

Could you please elaborate on this? If it's not the first request of an fbc package and the package is missing in the db, the deprecations shouldn't exist in the first place. If it's still present, it's a bug in the workflow somewhere which should be fixed rather than covering it.

Reply:
Discussed on a call and changes have been made to make the test more clear.